schema.json

```diff
@@ -0,0 +1,125 @@
+  "title": "Thunderbird In-App Notifications",
+  "type": "array",
+  "items": {
+    "type": "object",
+    "properties": {
+      "id": {
+        "type": "string",
+        "format": "uuid",
+        "description": "Unique ID set by the server"
+      },
+      "start_at": {
+        "type": "string",
+        "format": "date-time",
+        "description": "UTC Timestamp after which Thunderbird will show the event after startup."
+      },
+      "end_at": {
+        "type": "string",
+        "format": "date-time",
+        "description": "UTC Timestamp after which Thunderbird will never show the event, even if it has never been shown to the user."
+      },
+      "title": {
+        "type": "string",
+        "description": "Short sentence describing the event which will be displayed in the Thunderbird UI."
+      },
+      "description": {
+        "type": "string",
+        "description": "A short paragraph that can contain HTML and will be displayed in the Thunderbird UI."
+      },
+      "URL": {
+        "type": "string",
+        "format": "uri",
+        "description": "URL to open from the CTA, if any."
+      },
+      "CTA": {
+        "type": "string",
+        "description": "Link text to show for the URL."
+      },
+      "severity": {
+        "enum": [1, 2, 3, 4, 5],
+        "description": "Severity level, where 1 is the most important/urgent and 5 is the least."
+      },
+      "type": {
+        "enum": ["donation", "message", "security", "blog"],
+        "description": "Category of notification."
+      },
+      "targeting": {
+        "type": "object",
+        "properties": {
+          "percent_chance": {
+            "type": "number",
+            "minimum": 0,
+            "maximum": 100
+          },
+          "exclude": {
+            "type": "array",
+            "items": {
+              "$ref": "#/definitions/profile"
+            }
+          },
+          "include": {
+            "type": "array",
+            "items": {
+              "$ref": "#/definitions/profile"
+            }
+          }
+        }
+      }
+    },
+    "required": [
+      "id",
+      "start_at",
+      "end_at",
+      "title",
+      "description",
+      "severity",
+      "type",
+      "targeting"
+    ],
+    "dependentRequired": { "CTA": ["URL"] }
+  },
+  "definitions": {
+    "profile": {
+      "type": "object",
+      "properties": {
+        "locales": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "versions": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "channels": {
+          "type": "array",
+          "items": {
+            "enum": ["default", "esr", "release", "beta", "daily"],
+            "description": "Channel"
+          }
+        },
+        "operating_systems": {
+          "type": "array",
+          "items": {
+            "enum": [
+              "windows",
```

Comment:
I think these values should probably be the ones used by Thunderbird itself. That was my intent for all the targeting filters, that they mirror the values used internally in the client so Thunderbird doesn't need some crazy remapping scheme in `comm-central`.

I'm not sure what all possible values of %OS% are in Thunderbird. I think the major three are `WINNT`, `Darwin`, and `Linux` though. @jfx2006 might know or know where you can find lists of these things.

Reply:
I've been looking at https://searchfox.org/mozilla-central/source/toolkit/modules/AppConstants.sys.mjs since that's what I get at runtime.

Reply:
Ah, the URL formatter uses `appinfo`: https://searchfox.org/mozilla-central/source/toolkit/components/urlformatter/URLFormatter.sys.mjs#71-136, which uses `OS_TARGET`: https://searchfox.org/mozilla-central/source/toolkit/xre/nsAppRunner.cpp#1217, so that'd be the values you're talking about.

Reply:
> I've been looking at https://searchfox.org/mozilla-central/source/toolkit/modules/AppConstants.sys.mjs since that's what I get at runtime.

Ah, if it's easy for you to get that at runtime then nevermind, use that. Seems fine to me.

Reply:
If we'd want to match it that'd still mean  `"win"` and `"macosx"` as values though.

Reply:
Yeah definitely match, it will avoid having to maintain mappings on the client side.

Reply:
:+1: makes sense! I've updated those two enum values to match [AppConstants.sys.mjs](https://searchfox.org/mozilla-central/source/toolkit/modules/AppConstants.sys.mjs)